alchemiscale/tests/integration/storage/test_statestore.py

```diff
@@ -327,144 +327,201 @@ def test_get_tasks(self, n4js, network_tyk2, scope_test):
+        taskhub_sk: ScopedKey = n4js.create_taskhub(network_sk)
 
         # add a slightly different network
         an2 = AlchemicalNetwork(
             edges=list(an.edges)[:-1], name="tyk2_relative_benchmark_-1"
         )
         network_sk2 = n4js.create_network(an2, scope_test)
-        taskqueue_sk2: ScopedKey = n4js.create_taskqueue(network_sk2)
+        taskhub_sk2: ScopedKey = n4js.create_taskhub(network_sk2)
 
-        tq_sks: List[ScopedKey] = n4js.query_taskqueues()
+        tq_sks: List[ScopedKey] = n4js.query_taskhubs()
         assert len(tq_sks) == 2
         assert all([isinstance(i, ScopedKey) for i in tq_sks])
 
-        tq_dict: Dict[ScopedKey, TaskQueue] = n4js.query_taskqueues(return_gufe=True)
+        tq_dict: Dict[ScopedKey, TaskHub] = n4js.query_taskhubs(return_gufe=True)
         assert len(tq_dict) == 2
-        assert all([isinstance(i, TaskQueue) for i in tq_dict.values()])
+        assert all([isinstance(i, TaskHub) for i in tq_dict.values()])
 
     def test_action_task(self, n4js: Neo4jStore, network_tyk2, scope_test):
         an = network_tyk2
         network_sk = n4js.create_network(an, scope_test)
-        taskqueue_sk: ScopedKey = n4js.create_taskqueue(network_sk)
+        taskhub_sk: ScopedKey = n4js.create_taskhub(network_sk)
 
         transformation = list(an.edges)[0]
         transformation_sk = n4js.get_scoped_key(transformation, scope_test)
 
         # create 10 tasks
         task_sks = [n4js.create_task(transformation_sk) for i in range(10)]
 
-        # queue the tasks
-        n4js.action_tasks(task_sks, taskqueue_sk)
-
-        # count tasks in queue
-        queued_task_sks = n4js.get_taskqueue_tasks(taskqueue_sk)
-        assert task_sks == queued_task_sks
+        # action the tasks
+        n4js.action_tasks(task_sks, taskhub_sk)
 
+        # count tasks actioned
+        actioned_task_sks = n4js.get_taskhub_tasks(taskhub_sk)
+        assert set(task_sks) == set(actioned_task_sks)
+        assert len(task_sks) == 10
         # add a second network, with the transformation above missing
-        # try to add a task from that transformation to the new network's queue
+        # try to add a task from that transformation to the new network's hub
         # this should fail
+
         an2 = AlchemicalNetwork(
             edges=list(an.edges)[1:], name="tyk2_relative_benchmark_-1"
         )
         assert transformation not in an2.edges
 
         network_sk2 = n4js.create_network(an2, scope_test)
-        taskqueue_sk2: ScopedKey = n4js.create_taskqueue(network_sk2)
+        taskhub_sk2: ScopedKey = n4js.create_taskhub(network_sk2)
 
-        task_sks_fail = n4js.action_tasks(task_sks, taskqueue_sk2)
+        task_sks_fail = n4js.action_tasks(task_sks, taskhub_sk2)
         assert all([i is None for i in task_sks_fail])
 
-    def test_cancel_task(self, n4js, network_tyk2, scope_test):
+    def test_action_task_nested_fails(self, n4js: Neo4jStore, network_tyk2, scope_test):
```

Comment:
Not sure we really need this test; see comments on `action_tasks` check.

Reply:
Yep makes sense